Support whitespace in command path for gcp auth plugin

j3ffml opened the pull request:
```
External command option on gcp client auth plugin supports whitespace in command path.
```

Splitting on whitespace to get cmd+args breaks when the path the executable contains spaces. Resolve by adding a new "cmd-args" field to config to allow the full string of "cmd-path" to be interpreted as path to executable.

This change is backwards compatible with existing behavior.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/41653)
<!-- Reviewable:end -->

j3ffml committed:
Support whitespace in command path for gcp auth plugin

Specific use case is GKE users running gcloud/kubectl on Windows
with a cloud sdk installation path containing spaces. Also improving
test coverage using trick borrowed from exec_test.go

jlowdermilk commented:
I tagged this as do-not-merge as the fix doesn't appear to work on windows. May be a peculiarity of exec library on windows, needs some testing.

jlowdermilk commented:
https://github.com/kubernetes/kubernetes/pull/41653#issuecomment-283122707 was mistaken. I tested this on a windows vm, and it works, provided the .kube/config has been edited to split `cmd-args` from `cmd-path`.

cjcullen commented:
/lgtm

jlowdermilk commented:
re-adding lgtm. only change was rebase.

jlowdermilk commented:
flake: #42597
@k8s-bot kops aws e2e test this
@k8s-bot gci gce e2e test this

mengqiy reviewed (commented):
LGTM
